client/components/widget/query/query-result-data-service.js

```diff
@@ -233,19 +233,29 @@ QueryResultDataService.prototype.fetchResults = function(widget) {
         let rows = response.data.totalRows;
         let size = response.data.totalBytesProcessed;
         let time = response.data.elapsedTime;
-        let job = response.data.jobReference.jobId;
+        let jobReference = response.data.jobReference;
 
-        widget.state().datasource.job_id = job;
+        if (goog.isDefAndNotNull(jobReference)) {
```

Comment:
Then the jobid doesn't get recorded.  And there are a few edge error cases we've found where this occurs -- BigQuery returns, but there isn't a jobid.

At a larger scale, this class needs to be factored out for proper extensibility, but that hasn't been designed yet, and it will be in a separate PR.